contracts/FIFOMarket.sol

```diff
@@ -25,13 +23,13 @@ contract FIFOMarket is
   ERC1155HolderUpgradeable,
   IERC777RecipientUpgradeable
 {
-  using EnumerableSetUpgradeable for EnumerableSetUpgradeable.UintSet;
-
   IERC1820RegistryUpgradeable private _erc1820;
   Removal private _removal;
   Certificate private _certificate;
   NORI private _nori;
-  EnumerableSetUpgradeable.UintSet private _queue;
+  mapping(uint256 => uint256) private _queue;
+  uint256 private _queueHeadIndex;
+  uint256 private _queueNextInsertIndex;
```

Comment:
why do we need to store both `_queueHeadIndex` and `_queueNextInsertIndex`?

Reply:
discussed in OH about how we're using a map not array